packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart

```diff
@@ -18,13 +19,15 @@ import '../../src/test_flutter_command_runner.dart';
 
 void main() {
   FileSystem fileSystem;
+  ProcessManager processManager;
 
   setUpAll(() {
     Cache.disableLocking();
   });
 
   setUp(() {
     fileSystem = MemoryFileSystem.test();
+    processManager = FakeProcessManager.any();
```

Comment:
can you use `FakeProcessManager.empty()`? With `FakeProcessManager.any()`, it will accept any incoming requests to run sub-processes and always return 0. With `FakeProcessManager.empty()` and `FakeProcessManager.list()`, you need to register `FakeCommand`'s, which are expectations of commands that the Dart code under test should call. Here is an example small test file that uses it: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/general.shard/android/android_studio_validator_test.dart

This way if later on we regress the arguments to `dart format ...`, it will show up as a test failure.

Reply:
Then we'll need to migrate all tests since arb files paths are dynamic in common. Is that feasible?

Reply:
I think we've reached a conflicted condition.

For file system and context:
- If we still use `MemoryFileSystem.test()` to test, generated files won't be formatted since the dart binary artifact cannot be found.
- If we use `globals.fs` to test, https://github.com/flutter/flutter/pull/108563#discussion_r933739635 will be regressed, TL;DR `testUsingContext` is required.

For process manager:
- If we use `.empty` or `.list` to test, given the fact of only generated `.dart` files will be formatted, the command will be hard to construct.
- If we use `.any` to test, https://github.com/flutter/flutter/pull/108563#discussion_r942726787 will be regressed.

Reply:
Unless we make this opt-out.

Reply:
I'll go for `FakeProcessManager.empty()` for now.

Reply:
I'll go for `FakeProcessManager.empty()` for now.